stackstate_checks_base/stackstate_checks/base/checks/base.py

```diff
@@ -905,6 +906,15 @@ def _is_not_empty(self, field):
 
         return False
 
+    def get_check_state_path(self):
+        """
+        get_check_state_path returns the path where the check state is stored. By default the check configuration
+        location will be used. If state_location is set in the check configuration that will be used instead.
+        """
+        state_location = self.instance.get("state_location", self.get_agent_conf_d_path())
```

Comment:
@zandrewitte What will happen if the path doesn't exist or problem with the permission of writing? Is it already taken care?